### PR TITLE
Scale colorlegend to the scene resolution

### PR DESCRIPTION
**Pull request by asinghvi17**

### CODE REVIEW

**Comment by asinghvi17** [comment]
This is what the colormap looks like with this PR:
lol

File: src/basic_recipes/legend.jl [hunk 1]
```diff
@@ -34,10 +36,10 @@ creates a legend from a colormap
 """
 @recipe(ColorLegend, colormap, colorrange) do scene
     Theme(
-        width = (20, 200),
+        width = (20, lift(x -> x[2], theme(scene, :resolution))),
```
Comment (SimonDanisch) [inline src/basic_recipes/legend.jl]: should rather use `lift(x-> width(x)[2], pixelarea(scene))`.. The theme value won't update when resizing the window

Comment (asinghvi17) [reply src/basic_recipes/legend.jl]:
```suggestion
        width = (20, lift(x -> x[2], pixelarea(scene))),

File: src/basic_recipes/legend.jl [hunk 2]
```diff
@@ -34,10 +36,10 @@ creates a legend from a colormap
 """
 @recipe(ColorLegend, colormap, colorrange) do scene
     Theme(
-        width = (20, 200),
+        width = (20, lift(x -> x[2], pixelarea(scene))),
```
Comment (asinghvi17) [inline src/basic_recipes/legend.jl]:
```suggestion
        width = (20, lift(x -> x.h, pixelarea(scene))),
```

**Pull request merged**